MapboxNavigation/BottomBannerView.swift

```diff
@@ -77,6 +94,31 @@ open class BottomBannerView: UIView {
         arrivalTimeLabel.text = "10:09"
     }
     
+    @objc public func navigationService(_ service: NavigationService, didRerouteAlong route: Route, at location: CLLocation?, proactive: Bool) {
+        refreshETA()
+    }
+    
+    @objc public func navigationService(_ service: NavigationService, didUpdate progress: RouteProgress, with location: CLLocation, rawLocation: CLLocation) {
+        resetETATimer()
+        updateETA(routeProgress: progress)
+        previousProgress = progress
+    }
+    
+    @objc func removeTimer() {
+        timer?.invalidate()
+        timer = nil
+    }
+    
+    @objc func resetETATimer() {
+        removeTimer()
+        timer = Timer.scheduledTimer(timeInterval: 30, target: self, selector: #selector(refreshETA), userInfo: nil, repeats: true)
```

Comment:
I'm a little worried about using a scheduled timer here as it introduces a retain cycle. Previously, when owned by a view controller we broke the cycle on view disappearance, but now I'm worried we won't actually break the cycle if the only call to `removeTimer()` is in `deinit()`

Reply:
@akitchen It's worth pointing out that this code was moved (largely wholesale) from the [RouteMapViewController](https://github.com/mapbox/mapbox-navigation-ios/pull/1912/files#diff-292af689ce88d5b5aa06966456abc0e9L911). Do we want to change the mechanism of action here?

Reply:
yeabut the RouteMapViewController tore down the timer on view disappearance, which was more eager. it's that subtle difference that concerns me

would our DispatchTimer https://github.com/mapbox/mapbox-navigation-ios/blob/31e0742ad39fbb44842b20693135b190f9e00a60/MapboxCoreNavigation/DispatchTimer.swift be a better fit for this new factoring?